article.md

```diff
@@ -0,0 +1,333 @@
+# EIP712 authentication in cairo 
+In this article, I would like to show you how to verify messages signed as described in EIP712 on StarkNet. This requires recreating the original EIP712 payload, hashing it, and verifying Ethereum signature.
+## Background
+Make sure that you're familiar with [cairo-lang](https://www.cairo-lang.org) as it is essential in this case.
+There is a whole dedicated [piece](https://blog.swmansion.com/testing-starknet-contracts-made-easy-with-protostar-2ecdad3c9133) explaining almost everything you would need to know before reading this article, so make sure to check it out. Also, some knowledge about [EIP-712](https://eips.ethereum.org/EIPS/eip-712) would be nice too, as I won't be going into details about how it works.
+
+# Use case - creating a binding between ETH address and Starknet Address
+
+## Message structure
+```json
+{
+  "domain": {
+    "name": "TestContract",
+    "contractAddress": "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+    "version": "1"
+  },
+
+  "primaryType": "Payload",
+  "types": {
+    "domain": [
+      { "name": "name", "type": "string" },
+      { "name": "contractAddress", "type": "address"},
+      { "name": "version", "type": "string" }
+    ],
+    "Payload": [
+      { "name": "myStarkNetAddress", "type": "uint256" }
+    ]
+  }
+}
+```
+This is the structure that we'll be using in this example. Note that `domain` here represents `eip712Domain` which, after hashing, will become a domain separator that we will need. It ensures that the signature can only be used by our contract on the given chain, and that's what `contractAddress` is needed for (more on that later).<br/>
+`Payload` will be the main structure that we'll be constructing and signing. For the sake of simplicity, it only has one field but the EIP-712 standard allows for any type of structure.
```

Comment:
Maybe instead of "any type of structure" we could say "offers a lot of flexibility".